PublicMapping/districtbuilder
# Submit map button

## Overview

Adds workflow for submitting a map for a contest.

Two new fields have been added to `ProjectTemplate`:
 - `contestActive`: controls whether the "Submit" button is shown for maps connected to this template
 - `contestNextSteps`: HTML to display in the Next steps modal displayed after submission (optional, modal not shown if not set)
 
 Submission date is tracked on a new `Project.submittedDt` field, which also causes the submitted map to become read-only.

### Checklist


### Demo

Submitting a map w/o next steps HTML set:
![submit-no-next-steps](https://user-images.githubusercontent.com/4432106/159934876-4fd2d1ff-971f-4d90-82f9-21ddb24bfba8.gif)

Submitting a map with next steps HTML set:
![submit-next-steps](https://user-images.githubusercontent.com/4432106/159934924-fae92c0d-ad70-4633-8c76-d4b82fbba726.gif)

## Testing Instructions

- Use `scripts/load-dev-data` to reload your test data (it should make a contest active on the PA House of Representatives template for the Azavea organization), or use `scripts/dbshell` to set up contest fields for a project template of your choice.
- Create a map for that template
  - There should be a "Submit" button in the upper-right. Submitting your map should display a success toast and open the Next steps modal. Clicking the "Submitted" button should display a link to re-open the next steps modal.
  - Using `scripts/dbshell`, update your project template to set the next steps HTML to be empty: `UPDATE project_template SET contest_next_steps = '';`
  - Submit another map - you should still see the toast, but not the Next steps modal, and there should no longer be a link to it.
  - Using `scripts/dbshell`, update your project template to set the contest to be inactive: `UPDATE project_template SET contest_active = FALSE;`. You should still see the "Submitted" button for maps that were already submitted, but should no longer see a "Submit" button for maps that were not yet submitted.
- A few minutes after submitting a map, go to the Evaluate > Competitiveness detail. The "View on PlanScore" link should be present, not the "Upload to PlanScore" button.

Closes #1170
